src/main/java/com/thealgorithms/maths/ComplexNumberUtil.java

```diff
@@ -0,0 +1,264 @@
+     */
+    public static ComplexNumber subtract(ComplexNumber num1, ComplexNumber num2) {
+        return new ComplexNumber(num1.REAL - num2.REAL, num1.IMAGINARY - num2.IMAGINARY);
+    }
+
+    /**
+     * Multiplies two complex numbers.
+     *
+     * @param num1 the first complex number
+     * @param num2 the second complex number
+     * @return the product of num1 and num2
+     * @link <a href="https://en.wikipedia.org/wiki/Complex_number#Multiplication">...</a>
+     */
+    public static ComplexNumber multiply(ComplexNumber num1, ComplexNumber num2) {
+        return new ComplexNumber(num1.REAL * num2.REAL - num1.IMAGINARY * num2.IMAGINARY, num1.REAL * num2.IMAGINARY + num1.IMAGINARY * num2.REAL);
+    }
+
+    /**
+     * Divides the first complex number by the second complex number.
+     *
+     * @param num1 the numerator complex number
+     * @param num2 the denominator complex number
+     * @return the result of dividing num1 by num2
+     * @throws RuntimeException if the divisor (num2) is zero
+     * @link <a href="https://en.wikipedia.org/wiki/Complex_number#Complex_conjugate,_absolute_value_and_argument">...</a>
+     */
+    public static ComplexNumber divide(ComplexNumber num1, ComplexNumber num2) {
+        final double divisor = num2.REAL * num2.REAL + num2.IMAGINARY * num2.IMAGINARY;
+        if (divisor == 0) {
+            throw new RuntimeException("Cannot divide by zero");
+        }
+
+        return new ComplexNumber((num1.REAL * num2.REAL + num1.IMAGINARY * num2.IMAGINARY) / divisor, (num1.IMAGINARY * num2.REAL - num1.REAL * num2.IMAGINARY) / divisor);
+    }
+
+    /**
+     * Computes the absolute value (magnitude) of a complex number.
+     *
+     * @param num the complex number
+     * @return the absolute value of num
+     * @link <a href="https://en.wikipedia.org/wiki/Complex_number#Complex_conjugate,_absolute_value_and_argument">...</a>
+     */
+    public static double abs(ComplexNumber num) {
+        return Math.sqrt(num.REAL * num.REAL + num.IMAGINARY * num.IMAGINARY);
+    }
+
+    /**
+     * Computes the exponential function of a complex number.
+     *
+     * @param num the complex number
+     * @return e raised to the power of num
+     * @link <a href="https://en.wikipedia.org/wiki/Exponential_function#Continued_fractions_for_ex">...</a>
+     */
+    public static ComplexNumber exp(ComplexNumber num) {
+        final double coefficient = Math.exp(num.REAL);
+        return new ComplexNumber(coefficient * Math.cos(num.IMAGINARY), coefficient * Math.sin(num.IMAGINARY));
+    }
+
+    /**
+     * Computes the natural logarithm of a complex number.
+     *
+     * @param num the complex number
+     * @return the natural logarithm of num
+     * @throws RuntimeException if num is zero
+     * @link <a href="https://en.wikipedia.org/wiki/Complex_logarithm#Calculating_the_principal_value">...</a>
+     */
+    public static ComplexNumber ln(ComplexNumber num) {
```

Comment:
```suggestion
    public static ComplexNumber log(ComplexNumber num) {
```

Since you are computing the _principal value_, the name `ln` might be a bit misleading.